applications/chaineducation.md

```diff
@@ -0,0 +1,229 @@
+
+2. ![About Us Page](https://i.imgur.com/hNWkqOK.png  "About Us Page")
+
+3. ![Instructor Page](https://i.imgur.com/YcLxf2q.png  "Instructor Page")
+
+4. ![Tracks Page](https://i.imgur.com/8s4JwuB.png "Tracks Page")
+
+
+## Team :busts_in_silhouette:
+
+### Team members
+
+* Muskan Marwah, Co-Founder
+
+* Aviraj Khare, Co-Founder
+
+### Contact
+
+*  **Contact Name:** Aviraj Khare
+
+*  **Contact Email:** avirajkhare00@gmail.com
+
+
+### Legal Structure
+
+*  **Registered Legal Entity:** Not yet registered
+
+### Team's experience
+
+* Muskan Marwah: 6years flying experience as flight attendant. Started development in Rust and Solidity.
+
+* Aviraj Khare: Founder & CTO, ChainEducation, Ex Gojek, worked in web3 space from 2016 to 2019. Created a Decentralized exchange on top of the 0x protocol as a Whitelabel solution. From 2019-21 worked in the Gojek infrastructure team.
+
+### Team Code Repos
+
+* https://github.com/avirajkhare00
+* https://github.com/muskanmarwah1
+
+
+### Team LinkedIn Profiles
+
+* https://www.linkedin.com/in/avirajkhare00/
+
+## Development Roadmap :nut_and_bolt:
+
+# Overview
+
+## Grant level
+
+Level 2: Up to $30,000, 3 approvals
+
+
+## Development Roadmap :nut_and_bolt:
+
+*  **Total Estimated Duration:** 11 weeks
+
+*  **Total Costs:** 20,000 USD
+
+### Milestone 1
+
+*  **Estimated Duration:** 8 weeks
+
+*  **Costs:** 10,000 USD
+
+
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 0a. | Apache License 2.0 | All code will be published under Apache 2.0 |
+| 0b. | Documentation | We will provide both inline documentation of the code and a basic tutorial that explains how a user can (for example) deploy our smart contracts and send test transactions, which will show how the new functionality works. |
+| 0c. | Testing and it's Guide | Core functions will be fully covered by comprehensive unit tests written in Jest to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Articles | We will publish a series of tutorials along with online workshops within our discord community so that we can educate our users the use of this platform. |
+| 0. | Creation of smart contracts using Ink. | These smart contracts will be responsible to distribute NFTs to user once they complete the course. We will be using **erc721** token to mint the NFT: https://github.com/paritytech/ink/blob/master/examples/erc721/lib.rs. |
+| 1. | Creation of smart contracts using Ink. | Smart contract to store all the tracks. A track will have multiple courses. A course will have multiple lessons. Video of a lesson will go inside IPFS and it's hash will be stored inside a smart contract. A course will have multiple reviews. Smart contract will also store user's information weather user is an instructor or a student. |
```

Comment:
> A course will have multiple reviews.

Do only participants or finishers get to review? If so, wouldn't you need the registration/completion info also somewhere on-chain?

Reply:
Students who do finish the entire course get to review the course. We will be doing this to maintain the quality of the reviews, and we do need unbiased reviews. Yes. The wallet address of the user will be the user_id. Also, we will be storing additional information about the user, like education, current location, if a user is open to working, and skills. The page will look somewhat like the CV of a user.

Reply:
@semuelle Should I include all the data structures we will be using in the application?

Reply:
If you have them, please do share. But often, data structures change a lot during the development phase, so we don't ask for it. I am asking, because it's more useful if the review process was decentralised, i.e. the eligible reviewers' addresses on-chain. 

How does the NFT distribution contract work? What triggers it? How do you prove to the smart contract that a person has completed the course?

Reply:
> How does the NFT distribution contract work? What triggers it? How do you prove to the smart contract that a person has completed the course?

You asked a really nice question @semuelle. We will take a short quiz. If a person has completed a course, they should be able to answer 60% of the questions correctly. Also, the main smart contract will contain logic to execute Cross-Contract calling. It will mint new NFT.

> If you have them, please do share. But often, data structures change a lot during the development phase, so we don't ask for it. I am asking, because it's more useful if the review process was decentralised, i.e. the eligible reviewers' addresses on-chain.

The course review process is decentralized. I am sharing User structure below. Since change in struct is ever changing, I am including the struct here:
```
struct User {
    address: String,
    name: String,
    avatar_hash: String,
    courses_completed: u8,
    skills_id: Vec<u8>,
    location_id: u8,
    open_to_work: bool
}
```

Reply:
> You asked a really nice question @semuelle. We will take a short quiz. If a person has completed a course, they should be able to answer 60% of the questions correctly.

- That means that there must be some action happening in the backend, right? Something/Someone signals the main smart contract that user was successful? Or how do you prevent others from minting an NFT? 
- What happens if the user fails? Can they redo the quiz as many times as they like?

Reply:
> That means that there must be some action happening in the backend, right? Something/Someone signals the main smart contract that user was successful? Or how do you prevent others from minting an NFT?

Frontend JavaScript code will trigger the NFTDeployer smart contract. We won't be using backend anywhere. Only smart contract owner(we) will be able to mint the NFT.

> What happens if the user fails? Can they redo the quiz as many times as they like?

Users can try as many times as they want.

Reply:
> That means that there must be some action happening in the backend, right? Something/Someone signals the main smart contract that user was successful? Or how do you prevent others from minting an NFT?

Frontend JavaScript code will trigger the NFTDeployer smart contract. We won't be using backend anywhere. Only smart contract owner(we) will be able to mint the NFT.

> What happens if the user fails? Can they redo the quiz as many times as they like?

Users can try as many times as they want.

Reply:
I want to point out that we will be manually minting NFT for people who will be completing this course for some time. Eventually, we will automate this process too. The smart contract will have the addresses of people who have completed the course.